### PR TITLE
c'est quoi

### PR DESCRIPTION
**Pull request by Rebecca-auque**
expression familière

### CODE REVIEW

File: languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml [hunk 1]
```diff
@@ -10069,6 +10069,34 @@ though, and has since been slightly modified)-->
...
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id="C_EST_QUOI" name="c'est quoi" default="temp_off">
+          <rule>
+              <pattern>
+                  <marker>
+                      <token>c'</token>
+                      <token>est</token>
+                      <token>quoi</token>
+                  </marker>
+                  <token postag="D e s|D f s" postag_regexp="yes"/>
```
Comment (udomai) [inline languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml:10080]: `D [ef] s` sure would look nicer ;)

File: languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml [hunk 2]
```diff
@@ -10069,6 +10069,34 @@ though, and has since been slightly modified)-->
...
+                      <token>c'</token>
+                      <token>est</token>
+                      <token>quoi</token>
+                  </marker>
+                  <token postag="D e s|D m s" postag_regexp="yes"/>
+              </pattern>
+              <message>Cette expression peut être considérée comme familière.</message>
+              <suggestion>quel est</suggestion>
+              <example correction="quel est">Dis-moi <marker>c'est quoi</marker> ton nom.</example>
+          </rule>
```
Comment (udomai) [inline languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml:10098]: You always need to ask yourself "which obvious FPs is this pattern going to produce?"... even if the rule editor looks fine.

Problem with both rules:

- C'est quoi ce bordel ? → Quel est ce bordel ?
- C'est quoi ce bruit ? → Quel est ce bruit ?
- C'est quoi cette odeur ? → Quelle est cette odeur ?
- C'est quoi un adverbe ? → Quel est un adverbe ?
- C'est quoi une préposition ? → Quelle est une préposition ?
- C'est quoi le bonheur ? → Quel est le bonheur ?
- C'est quoi la philosophie ? → Quelle est la philosophie ?

And, by the way, making it case_sensitive or excluding SENT_START will not fix the problem. Users don't always start their sentences with an uppercase letter, and any of the above sentences might as well start with `Alors` or `Alors, `.

**Review by udomai — CHANGES_REQUESTED** [review]

File: languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml (hunk 2 above)
Comment (udomai) [reply languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml:10098]: Any feedback on this?

Comment (Rebecca-auque) [reply languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml]: Indeed, it was a tricky problem. I created subrules and I restricted the rule so it should be ready.

**Review by udomai — APPROVED** [review]

**Comment by udomai** [comment]
manually merged: 1d330451fe7c2b3e229358934884f4aba4f5247b

**Pull request closed without merging**